src/main/java/com/autotune/analyzer/plots/PlotManager.java

```diff
@@ -58,37 +57,46 @@ public PlotData.PlotsData generatePlots() {
             plotsDataMap.put(newTimestamp, new PlotData.PlotPoint(cpuUsage, memoryUsage));
             incrementStartTime = newTimestamp;
         }
 
         return new PlotData.PlotsData(recommendationTerm.getPlots_datapoints(), plotsDataMap);
     }
 
-    PlotData.UsageData getUsageData(Map<Timestamp, IntervalResults> resultInRange, AnalyzerConstants.MetricName metricName, String format) {
-        // Extract CPU values
-        List<Double> cpuValues = resultInRange.values().stream()
-                .filter(intervalResults -> intervalResults.getMetricResultsMap().containsKey(metricName))
-                .mapToDouble(intervalResults -> {
-                    MetricResults metricResults = intervalResults.getMetricResultsMap().get(metricName);
-                    return (metricResults != null && metricResults.getAggregationInfoResult() != null) ? metricResults.getAggregationInfoResult().getSum() : 0.0;
-                })
-                .boxed() // Convert double to Double
-                .collect(Collectors.toList());
-        if (cpuValues.size() > 0) {
-            double q1 = CommonUtils.percentile(TWENTYFIVE_PERCENTILE, cpuValues);
-            double q3 = CommonUtils.percentile(SEVENTYFIVE_PERCENTILE, cpuValues);
-            double median = CommonUtils.percentile(FIFTY_PERCENTILE, cpuValues);
+    PlotData.UsageData getUsageData(Map<Timestamp, IntervalResults> resultInRange, AnalyzerConstants.MetricName metricName) {
+        // stream through the results value and extract the CPU values
+        if (metricName.equals(AnalyzerConstants.MetricName.cpuUsage)) {
+            List<Double> cpuValues = CostBasedRecommendationModel.getCPUUsageList(resultInRange);
+            return getPercentileData(cpuValues, resultInRange, metricName);
+        } else {
+            // stream through the results value and extract the memory values
+            List<Double> memUsageList = resultInRange.values()
+                    .stream()
+                    .map(CostBasedRecommendationModel::calculateMemoryUsage)
+                    .toList();
+            return getPercentileData(memUsageList, resultInRange, metricName);
+        }
+    }
+
+    private PlotData.UsageData getPercentileData(List<Double> metricValues, Map<Timestamp, IntervalResults> resultInRange, AnalyzerConstants.MetricName metricName) {
+        if (!metricValues.isEmpty()) {
+            LOGGER.debug("metricValues : {}", metricValues);
+            double q1 = CommonUtils.percentile(TWENTYFIVE_PERCENTILE, metricValues);
+            double q3 = CommonUtils.percentile(SEVENTYFIVE_PERCENTILE, metricValues);
+            double median = CommonUtils.percentile(FIFTY_PERCENTILE, metricValues);
             // Find max and min
-            double max = Collections.max(cpuValues);
-            double min = Collections.min(cpuValues);
+            double max = Collections.max(metricValues);
+            double min = Collections.min(metricValues);
```

Comment:
Is min value being taken cpu_min and memory_min metrics ?

Reply:
As discussed, this has been taken into consideration now.